dbt/include/dremio/macros/materializations/incremental/incremental.sql

```diff
@@ -14,64 +14,96 @@ limitations under the License.*/
 
 {% materialization incremental, adapter='dremio' -%}
 
-  {%- set identifier = model['alias'] -%}
-  {%- set old_relation = adapter.get_relation(database=database, schema=schema, identifier=identifier) -%}
-  {%- set target_relation = api.Relation.create(identifier=identifier,
-                                                schema=schema,
-                                                database=database,
-                                                type='table') -%}
-
-  {%- set tmp_identifier = model['name'] + '__dbt_tmp' -%}
-  {%- set tmp_relation = adapter.get_relation(database=database, schema=schema, identifier=tmp_identifier) -%}
-  {%- set target_tmp_relation = api.Relation.create(identifier=tmp_identifier,
-                                                schema=schema,
-                                                database=database,
-                                                type='table') -%}
-
-  {%- set raw_strategy = config.get('incremental_strategy', validator=validation.any[basestring]) or 'append' -%}
-  {%- set raw_file_format = config.get('format', validator=validation.any[basestring]) or 'iceberg' -%}
-  {%- set file_format = dbt_dremio_validate_get_file_format(raw_file_format) -%}
-  {%- set strategy = dbt_dremio_validate_get_incremental_strategy(raw_strategy, file_format) -%}
+  -- relations
+  {%- set existing_relation = load_cached_relation(this) -%} 
+  {%- set target_relation = this.incorporate(type='table') -%}

```

Comment:
We should be careful about using dbt's defaults for these macros. In the end, we as a team need to add more dremio specific test cases around custom schemas and different targets, to ensure changes like these don't break any of the edge cases.

Reply:
On the contrary, I think incorporating new dbt logic is necessary. The new dbt logic is more elegant and less "hacky". It's also not like we're just using dbt's default, we're using its logic but keeping Dremio specific changes. In the future I think the ideal direction is actually using dbt's defaults for everything and just having Dremio specific macros. That way we do not have to maintain any of these materialization strategies - that will be done by the dbt team.

However, I agree we should have more Dremio specific test cases.